src/datasets/vtr_dataset.py

```diff
@@ -55,3 +54,59 @@ def collate_function(self, batch: list[tuple[torch.Tensor, int]]) -> dict[str, t
             "attention_mask": attention_mask,
             "labels": torch.tensor(labels, dtype=torch.float),
         }
+
+
+class VTRDatasetOCR(Dataset):
+    def __init__(
+        self,
+        labeled_texts: list[DatasetSample],
+        font: str,
+        font_size: int = 15,
+        window_size: int = 30,
+        stride: int = 5,
+        max_seq_len: int = 512,
+        ratio: float = 0.7,
+    ):
+        logger.info(f"Initializing VTRDatasetOCR with {len(labeled_texts)} samples, use max seq len {max_seq_len}")
+        self.labeled_texts = labeled_texts
+        self.max_seq_len = max_seq_len
+
+        self.slicer = VTRSlicerWithText(
+            font=font, font_size=font_size, window_size=window_size, stride=stride, ratio=ratio
+        )
+
+    def __len__(self) -> int:
+        return len(self.labeled_texts)
+
+    def __getitem__(self, index) -> tuple[torch.Tensor, int, list[str]]:
+        sample = self.labeled_texts[index]
+        raw_text = clean_text(sample["text"])
+
+        # [n slices; font size; window size]
+        slices, slice_text = self.slicer(raw_text)
+        slices = slices[: self.max_seq_len]
+        slice_text = slice_text[: self.max_seq_len]
+        return slices, sample["label"], slice_text
+
+    def collate_function(
```

Comment:
This collate function is quite the same as in `VTRDataset`. I suggest inheriting this class from `VTRDataset`. Then
1. There is no need for the `__len__` method
2. Collate function would be a call to the super method and then just adding text field (3-4 lines)